### ISSUE TITLE
Can we lowercase build directories?

### ISSUE THREAD

**Issue by bartbroere** [pyodide/pyodide#2154]
## Proposed refactoring or deprecation
Lowercase all the (packages) build directories by default.

### Motivation
Windows filesystems don't distinguish well enough between an upper case and lower case path.

### Pitch
My development setup (sometimes) is a Windows setup, with the Pyodide repository volume mounted into a Docker container running `pyodide-env`. When building and developing packages, I'm getting errors when the build process is trying to rename this. If all build directories would be lowercased by default, this error probably wouldn't occur.

### Additional context
Attached a version of this error.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/src/pyodide-build/pyodide_build/__main__.py", line 62, in <module>
    main()
  File "/src/pyodide-build/pyodide_build/__main__.py", line 56, in main
    args.func(args)
  File "/src/pyodide-build/pyodide_build/buildpkg.py", line 821, in main
    build_package(
  File "/src/pyodide-build/pyodide_build/buildpkg.py", line 643, in build_package
    prepare_source(pkg_root, build_dir, srcpath, source_metadata)
  File "/src/pyodide-build/pyodide_build/buildpkg.py", line 253, in prepare_source
    download_and_extract(buildpath, srcpath, src_metadata)
  File "/src/pyodide-build/pyodide_build/buildpkg.py", line 216, in download_and_extract
    shutil.move(buildpath / extract_dir_name, srcpath)
  File "/usr/local/lib/python3.9/shutil.py", line 796, in move
    os.rename(src, dst)
OSError: [Errno 39] Directory not empty: '/src/packages/pillow/build/Pillow-9.0.0' -> '/src/packages/pillow/build/pillow-9.0.0'
[2022-02-02 13:56:16] Building package pillow...
[2022-02-02 13:56:38] Failed building package pillow in 21.9 seconds.
```

**Comment by hoodmane** [comment]
Thanks for the report! 

I want them to be lower case too, the problem is that the upper case version came out of a .tar.gz archive. We could try catching and throwing away the error but that feels weird.

Could you get windows subsystem Linux and mount the docker image into a folder inside wsl?

**Comment by ryanking13** [comment]
Hmm... something is clearly broken due to the cross filesystem operation.

**Comment by hoodmane** [comment]
As I said on #2212:

> The best solution is to move the Pyodide stuff into the Linux file system, say into `~/pyodide`. This has the additional benefit that all file system operations will be waaay faster so it should improve build times by a large amount.

_Originally posted by @hoodmane in https://github.com/pyodide/pyodide/issues/2212#issuecomment-1049926314_

But I will say this is closed by #2278.
